common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala

```diff
@@ -3326,6 +3327,15 @@ object CelebornConf extends Logging {
       .booleanConf
       .createWithDefault(false)
 
+  val SLOT_ASSIGN_MAX_WORKERS: ConfigEntry[Int] =
```

Comment:
1.According to our configuration key rule, celeborn.client can only for client side, so IMO better add another config key for master.

2. When we add this configuration, It would potentially reduce app worker resources for exist cluster work loads, then it may impact some app performance, so i perfer we can enable  this feature at client side or both side to keep existing behavior 


3. As client's configuration priority is higher than master, could we add a max limit side(10000) in master, (Say a large cluster say 100000 workers), to avoid client use so many workers.

Reply:
Thanks @RexXiong for your comment!
1. I have separated into `celeborn.master.slot.assign.maxWorkers` and `celeborn.client.slot.assign.maxWorkers`
2. I have changed the default to 10000, so I think it's big enough for a single shuffle.
3. I changed the logic to always choose the smaller positive configs from client and master, this way master's config acts as a limit.